Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,372 @@
+  -- synthesize an argument `[IsFiniteMeasure (μ.restrict E)]`.
+  -- It is enough to have in the context a term of `Fact (μ E < ∞)`, which is our `ffmE` below,
+  -- which is automatically fed into `Restrict.isFiniteInstance`.
+  have ffmE : Fact _ := { out := hfmE }
+  have hInner := tendsto_Lp_of_tendsto_ae_of_meas hp hp' hf hg hgE' huiE hfgE
+  rw [ENNReal.tendsto_atTop_zero] at hInner
+  -- get a sufficiently large N for given ε, and consider any n ≥ N
+  obtain ⟨N, hfngε⟩ := hInner (ε / 3) hε'
+  use N; intro n hn
+  -- get interior estimates
+  have hmfngE : AEStronglyMeasurable _ μ := (((hf n).sub hg).indicator hmE).aestronglyMeasurable
+  have hfngEε := calc
+    snorm (E.indicator (f n - g)) p μ
+      = snorm (f n - g) p (μ.restrict E) := snorm_indicator_eq_snorm_restrict hmE
+    _ ≤ ε / 3                            := hfngε n hn
+  -- get exterior estimates
+  have hmgEc : AEStronglyMeasurable _ μ := (hg.indicator hmE.compl).aestronglyMeasurable
+  have hgEcε := calc
+    snorm (Eᶜ.indicator g) p μ
+      ≤ snorm (Efᶜ.indicator (Egᶜ.indicator g)) p μ := by
+        unfold_let E; rw [compl_union, ← indicator_indicator]
+    _ ≤ snorm (Egᶜ.indicator g) p μ := snorm_indicator_le _
+    _ ≤ ε / 3 := hgε.le
+  have hmfnEc : AEStronglyMeasurable _ μ := ((hf n).indicator hmE.compl).aestronglyMeasurable
+  have hfnEcε : snorm (Eᶜ.indicator (f n)) p μ ≤ ε / 3 := calc
+    snorm (Eᶜ.indicator (f n)) p μ
+      ≤ snorm (Egᶜ.indicator (Efᶜ.indicator (f n))) p μ := by
+        unfold_let E; rw [compl_union, inter_comm, ← indicator_indicator]
+    _ ≤ snorm (Efᶜ.indicator (f n)) p μ := snorm_indicator_le _
+    _ ≤ ε / 3 := hfε n
+  have hmfngEc : AEStronglyMeasurable _ μ :=
+    (((hf n).sub hg).indicator hmE.compl).aestronglyMeasurable
+  have hfngEcε := calc
+    snorm (Eᶜ.indicator (f n - g)) p μ
+      = snorm (Eᶜ.indicator (f n) - Eᶜ.indicator g) p μ := by
+        rw [(Eᶜ.indicator_sub' _ _)]
+    _ ≤ snorm (Eᶜ.indicator (f n)) p μ + snorm (Eᶜ.indicator g) p μ := by
+        apply snorm_sub_le (by assumption) (by assumption) hp
+    _ ≤ ε / 3 + ε / 3 := add_le_add hfnEcε hgEcε
+  -- finally, combine interior and exterior estimates
+  calc
+    snorm (f n - g) p μ
+      = snorm (Eᶜ.indicator (f n - g) + E.indicator (f n - g)) p μ := by
+        congr; exact (E.indicator_compl_add_self _).symm
+    _ ≤ snorm (indicator Eᶜ (f n - g)) p μ + snorm (indicator E (f n - g)) p μ := by
+        apply snorm_add_le (by assumption) (by assumption) hp
+    _ ≤ (ε / 3 + ε / 3) + ε / 3 := add_le_add hfngEcε hfngEε
+    _ = ε := by simp only [ENNReal.add_thirds] --ENNReal.add_thirds ε
+
+/- Lemma used in `tendsto_Lp_notFinite_of_tendsto_ae`.
+   XXX: Alternative name: `ae_tendsto_ae_congr`? -/
+theorem tendsto_ae_congr_ae {f f' : ℕ → α → β} {g g' : α → β}
+    (hff' : ∀ (n : ℕ), f n =ᵐ[μ] f' n) (hgg' : g =ᵐ[μ] g')
+    (hfg : ∀ᵐ x ∂μ, Tendsto (fun n => f n x) atTop (𝓝 (g x))) :
+    ∀ᵐ x ∂μ, Tendsto (fun n => f' n x) atTop (𝓝 (g' x)) := by
```

Comment:
This has nothing to do with `UnifTight`, right? Can you move it to an earlier file? unless it's supposed to be an auxiliary lemma? In that case, please make it `private`.

Reply:
OK, not sure where else it would fit. I will make it `private`.